packages/vuetify/src/components/VDatePicker/VDatePickerDateTable.ts

```diff
@@ -48,29 +52,21 @@ export default mixins(
     genTHead () {
       const days = this.weekDays.map(day => this.$createElement('th', day))
       this.showWeek && days.unshift(this.$createElement('th'))
+
       return this.$createElement('thead', this.genTR(days))
     },
     // Returns number of the days from the firstDayOfWeek to the first day of the current month
     weekDaysBeforeFirstDayOfTheMonth () {
       const firstDayOfTheMonth = new Date(`${this.displayedYear}-${pad(this.displayedMonth + 1)}-01T00:00:00+00:00`)
       const weekDay = firstDayOfTheMonth.getUTCDay()
+
       return (weekDay - parseInt(this.firstDayOfWeek) + 7) % 7
     },
-    getWeekNumber () {
-      let dayOfYear = [0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334][this.displayedMonth]
-      if (this.displayedMonth > 1 &&
-        (((this.displayedYear % 4 === 0) && (this.displayedYear % 100 !== 0)) || (this.displayedYear % 400 === 0))
-      ) {
-        dayOfYear++
-      }
-      const offset = (
-        this.displayedYear +
-        ((this.displayedYear - 1) >> 2) -
-        Math.floor((this.displayedYear - 1) / 100) +
-        Math.floor((this.displayedYear - 1) / 400) -
-        Number(this.firstDayOfWeek)
-      ) % 7 // https://en.wikipedia.org/wiki/Zeller%27s_congruence
-      return Math.floor((dayOfYear + offset) / 7) + 1
+    getWeekNumber (dayInMonth: number) {
+      console.log(parseInt(this.firstDayOfYear))
+
+      return determineWeeknumber(this.displayedYear, this.displayedMonth, dayInMonth,
```

Comment:
This is unreadable, break up into new lines

Reply:
Every argument is now on a single line

Reply:
Not really, there are 5 arguments but only 2 lines